packages/chain-adapters/src/dogecoin/DogecoinChainAdapter.ts

```diff
@@ -0,0 +1,458 @@
+    this.coinName = args.coinName
+    this.assetId = toAssetId({
+      chainId: this.chainId,
+      assetNamespace: 'slip44',
+      assetReference: ASSET_REFERENCE.Dogecoin
+    })
+    console.info(`DogecoinChainAdapter 4: ${this.assetId}`)
+    this.parser = new unchained.bitcoin.TransactionParser({ chainId: this.chainId })
+    console.info(`DogecoinChainAdapter 5`)
+  }
+
+  getDisplayName() {
+    return 'Dogecoin'
+  }
+
+  getType(): KnownChainIds.DogecoinMainnet {
+    return KnownChainIds.DogecoinMainnet
+  }
+
+  getFeeAssetId(): AssetId {
+    return 'bip122:00000000001a91e3dace36e2be3bf030/slip44:3'
+  }
+
+  getSupportedAccountTypes() {
+    return ChainAdapter.supportedAccountTypes
+  }
+
+  async getTxHistory(
+    input: TxHistoryInput
+  ): Promise<TxHistoryResponse<KnownChainIds.DogecoinMainnet>> {
+    if (!this.accountAddresses[input.pubkey]) {
+      await this.getAccount(input.pubkey)
+    }
+
+    const { data } = await this.providers.http.getTxHistory({
+      pubkey: input.pubkey,
+      pageSize: input.pageSize,
+      cursor: input.cursor
+    })
+
+    const getAddresses = (tx: unchained.bitcoin.BitcoinTx): Array<string> => {
+      const addresses: Array<string> = []
+
+      tx.vin?.forEach((vin) => {
+        if (!vin.addresses) return
+        addresses.push(...vin.addresses)
+      })
+
+      tx.vout?.forEach((vout) => {
+        if (!vout.addresses) return
+        addresses.push(...vout.addresses)
+      })
+
+      return [...new Set(addresses)]
+    }
+
+    const txs = await Promise.all(
+      (data.txs ?? []).map(async (tx) => {
+        const addresses = getAddresses(tx).filter((addr) =>
+          this.accountAddresses[input.pubkey].includes(addr)
+        )
+
+        return await Promise.all(
+          addresses.map(async (addr) => {
+            const parsedTx = await this.parser.parse(tx, addr)
+
+            return {
+              address: addr,
+              blockHash: parsedTx.blockHash,
+              blockHeight: parsedTx.blockHeight,
+              blockTime: parsedTx.blockTime,
+              chainId: parsedTx.chainId,
+              chain: this.getType(),
+              confirmations: parsedTx.confirmations,
+              txid: parsedTx.txid,
+              fee: parsedTx.fee,
+              status: getStatus(parsedTx.status),
+              tradeDetails: parsedTx.trade,
+              transfers: parsedTx.transfers.map((transfer) => ({
+                assetId: transfer.assetId,
+                from: transfer.from,
+                to: transfer.to,
+                type: getType(transfer.type),
+                value: transfer.totalValue
+              }))
+            }
+          })
+        )
+      })
+    )
+
+    return {
+      cursor: data.cursor ?? '',
+      pubkey: input.pubkey,
+      transactions: txs.flat()
+    }
+  }
+
+  async buildSendTransaction(tx: BuildSendTxInput<KnownChainIds.DogecoinMainnet>): Promise<{
```

Comment:
i recently moved `buildSendTransaction` to the EVMBaseAdapter as eth mainnet/avalanche were identical - wonder if we can do the same here too?